datasketches-memory-java8-tests/src/test/java/org/apache/datasketches/memory/test/LeafImplTest.java

```diff
@@ -30,91 +32,92 @@
 import java.nio.ByteBuffer;
 import java.nio.ByteOrder;
 
-import org.apache.datasketches.memory.WritableHandle;
-import org.apache.datasketches.memory.internal.Util;
+import org.apache.datasketches.memory.MemoryRequestServer;
 import org.apache.datasketches.memory.WritableBuffer;
-
-import org.apache.datasketches.memory.WritableMemory;
+import org.apache.datasketches.memory.WritableHandle;
 import org.apache.datasketches.memory.WritableMapHandle;
+import org.apache.datasketches.memory.WritableMemory;
 import org.testng.annotations.Test;
 
 /**
  * @author Lee Rhodes
  */
 @SuppressWarnings("javadoc")
 public class LeafImplTest {
-  static final ByteOrder LE = ByteOrder.LITTLE_ENDIAN;
-  static final ByteOrder BE = ByteOrder.BIG_ENDIAN;
+  private static final ByteOrder NO = nativeByteOrder;
+  private static final ByteOrder NNO = nonNativeByteOrder;
+  private static final MemoryRequestServer dummyMemReqSvr = new DummyMemoryRequestServer();
+  
+  private static ByteOrder otherOrder(ByteOrder order) { return (order == NO) ? NNO : NO; } 
+  
+  static class DummyMemoryRequestServer implements MemoryRequestServer {
+    @Override
+    public WritableMemory request(long capacityBytes) { return null; }
+    @Override
+    public void requestClose(WritableMemory memToClose, WritableMemory newMemory) { }
+  }
   
   @Test
   public void checkDirectLeafs() throws Exception {
     long off = 0;
     long cap = 128;
-    try (WritableHandle wdh = WritableMemory.allocateDirect(cap)) {
-      WritableMemory memLE = wdh.getWritable();
-      memLE.putShort(0, (short) 1);
-      checkDirect(memLE, off, cap);
+    // Off Heap, Native order, No ByteBuffer, has MemReqSvr
+    try (WritableHandle wdh = WritableMemory.allocateDirect(cap, NO, dummyMemReqSvr)) {
+      WritableMemory memNO = wdh.getWritable();
+      memNO.putShort(0, (short) 1);
+      assertNull(ReflectUtil.getUnsafeObject(memNO));
+      assertTrue(memNO.isDirect());
+      checkCombinations(memNO, off, cap, memNO.isDirect(), NO, false, true);
+    }
+    // Off Heap, Non Native order, No ByteBuffer, has MemReqSvr
+    try (WritableHandle wdh = WritableMemory.allocateDirect(cap, NNO, dummyMemReqSvr)) {
+      WritableMemory memNNO = wdh.getWritable();
+      memNNO.putShort(0, (short) 1);
+      assertNull(ReflectUtil.getUnsafeObject(memNNO));
+      assertTrue(memNNO.isDirect());
+      checkCombinations(memNNO, off, cap, memNNO.isDirect(), NNO, false, true);
```

Comment:
Is the result of ReflectUtil.getUnsafeObject null in both cases because the memory is off-heap?

Reply:
Yes.